### DISCUSSION TITLE
Failing to re-connect to redis cluster deployed as a k8s service when rediscluster service is restarted

### DISCUSSION THREAD

**Discussion by pratikgithub** [redis/lettuce#2362] [Q&A] [answered]
I have a redis cluster with 6 nodes (3 masters and 3 slaves) deployed as k8s service. 
Its resultant redis URL is "redis://rediscluster:6379". 

Consuming application uses Lettuce as a redis client and can connect to redis successfully.

But when the redis service is redeployed or any node is restarted from the cluster, Lettuce is failing with "Unable to connect" exception.

Error- Cannot reconnect to [172.16.32.178/<unresolved>:6379]: Connection reset\njava.net.SocketException: Connection reset\n\tat 

I am checking _connection.isOpen() flag to check the connection's liveness. 
I debugged and observed that "redis://rediscluster:6379" gets resolved to some ip address and Lettuce connects to that ip address.

Actually when redis cluster service restarts by any reason, its ip address changes but Lettuce still tries to connect to the old ip.

I am looking for any Lettuce's API with which we can reset its connection memory when the redis cluster service is redeployed/restarted and existing connection is not valid ( isOpen() = false).

Requesting help on this .

**Comment by mp911de** [comment]
If you configure a custom `DnsResolver` without caching, then you should be able to obtain a freshly resolved address on each resolution attempt.

**Comment by pratikgithub** [comment, marked as answer]
This code worked.

```
      ClusterTopologyRefreshOptions topologyRefreshOptions = ClusterTopologyRefreshOptions.builder()
                    .enablePeriodicRefresh(Duration.ofSeconds(30))
                    .enableAllAdaptiveRefreshTriggers()
                    .build();

      ClusterClientOptions clusterClientOptions = ClusterClientOptions.builder()
              .topologyRefreshOptions(topologyRefreshOptions)
              .autoReconnect(true)
              .disconnectedBehavior(ClientOptions.DisconnectedBehavior.ACCEPT_COMMANDS)
              .timeoutOptions(TimeoutOptions.builder().build())
              .maxRedirects(DEFAULT_MAX_REDIRECTS * 2)
              .build();
              
              AbstractRedisClient  _client;
              _client = RedisClusterClient.create(url);
              _client.setDefaultTimeout(RedisURI.DEFAULT_TIMEOUT_DURATION);
              ((RedisClusterClient) _client).setOptions(clusterClientOptions);
```
